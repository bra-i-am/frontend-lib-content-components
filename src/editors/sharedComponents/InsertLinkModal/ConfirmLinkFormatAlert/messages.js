import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  confirmLinkFormatAlertTitle: {
    id: 'confirm.link.format.alert.title',
    defaultMessage: 'URL format',
    description: 'Title message format link alert when the URL does not have a protocol',
  },
  confirmLinkFormatAlertDescription: {
    id: 'confirm.link.format.alert.description',
    defaultMessage: 'The URL you entered seems to be an external link. Do you want to add the required prefix?',
    description: 'Description message format link alert when the URL does not have a protocol',
  },
  confirmLinkFormatAlertConfirm: {
    id: 'confirm.link.format.alert.confirm',
    defaultMessage: 'Confirm',
    description: 'Confirm button alert when the URL does not have a protocol',
  },
  confirmLinkFormatAlertCancel: {
    id: 'confirm.link.format.alert.cancel',
    defaultMessage: 'Cancel',
    description: 'Cancel button alert when the URL does not have a protocol',
  },
  confirmLinkFormatAlertHttpProtocol: {
    id: 'confirm.link.format.alert.http.protocol',
    defaultMessage: 'http',
    description: 'http option in the alert when the URL does not have a protocol',
  },
  confirmLinkFormatAlertHttpsProtocol: {
    id: 'confirm.link.format.alert.https.protocol',
    defaultMessage: 'https',
    description: 'https option in the alert the URL does not have a protocol',
  },
});

export default messages;
