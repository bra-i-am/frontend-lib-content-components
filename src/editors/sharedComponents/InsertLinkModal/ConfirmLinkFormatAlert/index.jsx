import { useState } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from '@edx/frontend-platform/i18n';
import {
  Form,
  AlertModal,
  ActionRow,
  Button,
} from '@openedx/paragon';
import { Warning } from '@openedx/paragon/icons';

import messages from './messages';

const ConfirmLinkFormatAlert = ({
  url,
  onCloseAlert,
  editorRef,
}) => {
  const intl = useIntl();
  const [protocol, setProtocol] = useState('');
  const handleChangeProtocol = e => setProtocol(e.target.value);

  const handleCancel = () => onCloseAlert();

  /* istanbul ignore next */
  const handleConfirm = () => {
    const editor = editorRef.current;
    if (editor) {
      const selectedText = editor.selection.getContent({ format: 'text' });
      const urlPath = `${protocol}${url}`;

      if (selectedText.trim() !== '') {
        const linkHtml = `<a href="${urlPath}" data-mce-href="${urlPath}" target="_blank">${selectedText}</a>`;
        editor.selection.setContent(linkHtml);
      }
    }
    onCloseAlert();
  };

  return (
    <AlertModal
      title={intl.formatMessage(messages.confirmLinkFormatAlertTitle)}
      isOpen
      onClose={onCloseAlert}
      variant="warning"
      icon={Warning}
      footerNode={(
        <ActionRow>
          <Button
            variant="tertiary"
            onClick={handleCancel}
            data-testid="cancel-button"
          >
            {intl.formatMessage(messages.confirmLinkFormatAlertCancel)}
          </Button>
          <Button
            variant="primary"
            disabled={!protocol}
            data-testid="confirm-button"
            onClick={handleConfirm}
          >
            {intl.formatMessage(messages.confirmLinkFormatAlertConfirm)}
          </Button>
        </ActionRow>
      )}
    >
      <Form.Group>
        <Form.Label className="my-4">
          {intl.formatMessage(messages.confirmLinkFormatAlertDescription)}
        </Form.Label>
        <Form.RadioSet
          name="colors"
          onChange={handleChangeProtocol}
          value={protocol}
        >
          <Form.Radio value="http://" data-testid="radio-http">
            {intl.formatMessage(messages.confirmLinkFormatAlertHttpProtocol)}
          </Form.Radio>
          <Form.Radio value="https://" data-testid="radio-https">
            {intl.formatMessage(messages.confirmLinkFormatAlertHttpsProtocol)}
          </Form.Radio>
        </Form.RadioSet>
      </Form.Group>
    </AlertModal>
  );
};

ConfirmLinkFormatAlert.propTypes = {
  url: PropTypes.string.isRequired,
  onCloseAlert: PropTypes.func.isRequired,
  editorRef: PropTypes.shape({
    current: PropTypes.shape({
      selection: PropTypes.shape({
        getContent: PropTypes.func,
        setContent: PropTypes.func,
      }),
    }),
  }).isRequired,
};

export default ConfirmLinkFormatAlert;
