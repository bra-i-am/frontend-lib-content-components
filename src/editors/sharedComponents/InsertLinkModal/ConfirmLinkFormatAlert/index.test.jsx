import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { fireEvent, render, screen } from '@testing-library/react';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import ConfirmLinkFormatAlert from '.';

jest.unmock('@edx/frontend-platform/i18n');
jest.unmock('@openedx/paragon');
jest.unmock('@openedx/paragon/icons');

describe('ConfirmLinkFormatAlert', () => {
  const mockProps = {
    url: 'www.example.com',
    onCloseAlert: jest.fn(),
    editorRef: {
      current: {
        selection: {
          getContent: jest.fn(),
          setContent: jest.fn(),
        },
      },
    },
  };
  // eslint-disable-next-line react/prop-types
  const IntlProviderWrapper = ({ children }) => (
    <IntlProvider locale="en">{children}</IntlProvider>
  );

  beforeEach(() => {
    jest.clearAllMocks();
  });

  const renderComponent = (overrideProps = {}) => render(
    <IntlProviderWrapper>
      <ConfirmLinkFormatAlert {...mockProps} {...overrideProps} />
    </IntlProviderWrapper>,
  );

  test('snapshot', async () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  test('renders without crashing', async () => {
    renderComponent();
    expect(screen.getByText('URL format')).toBeInTheDocument();
  });

  test('clicking cancel button calls onCloseAlert', () => {
    renderComponent();
    fireEvent.click(screen.getByTestId('cancel-button'));
    expect(mockProps.onCloseAlert).toHaveBeenCalled();
  });

  test('confirm button is disabled when protocol is not selected', () => {
    renderComponent();
    const confirmButton = screen.getByTestId('confirm-button');
    expect(confirmButton).toBeDisabled();
  });

  test('confirm button is enabled when protocol is selected', () => {
    renderComponent();
    fireEvent.click(screen.getByTestId('radio-http'));
    const confirmButton = screen.getByTestId('confirm-button');
    expect(confirmButton).not.toBeDisabled();
  });
});
