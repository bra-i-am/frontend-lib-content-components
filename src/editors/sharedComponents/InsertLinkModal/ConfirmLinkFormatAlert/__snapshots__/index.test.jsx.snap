// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmLinkFormatAlert snapshot 1`] = `
<div
  aria-hidden="true"
  data-focus-on-hidden="true"
/>
`;
